lib/BlockingDriver.php

```diff
@@ -358,8 +367,14 @@ public function touch(string $path, int $time = null, int $atime = null): Promis
     public function get(string $path): Promise
     {
         $result = @\file_get_contents($path);
+        if ($result === false) {
+            $message = 'Could not open the file.';
+            if ($error = \error_get_last()) {
+                $message .= \sprintf(" Errno: %d; %s", $error["type"], $error["message"]);
+            }
+        }
         return ($result === false)
-            ? new Failure(new FilesystemException(\error_get_last()["message"]))
+            ? new Failure(new FilesystemException($message))
```

Comment:
If there's an if, we don't need the ternary anymore.

Reply:
OK, fixed.
This, and the other two PRs I made, contain fixes that allow amphp libraries to run in a restricted environment (PHP safe mode, i.e. plain webhosts).

In this case, if the amphp/dns library fails to fetch /etc/resolv.conf, and error_get_last returns null, PHP throws a type error when null (instead of a string) is passed to the constructor of FilesystemException.

Merry XMas btw!

Reply:
But is it empty, because the error reporting is off?

Reply:
Error reporting is set to E_ALL.

Reply:
What triggers the failure then? The function simply returns false without any warning?

Reply:
Oh nevermind, for some reason I returned true instead of false in my custom error handler, that's why error info could not be fetched.
But I suppose it's OK to merge this, anyway: if the user's error handler is messed up, at least a proper exception will be returned, instead of a weird type error.

Reply:
I guess in that case we should simply replace the error handler temporarily.